akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala

```diff
@@ -225,8 +223,12 @@ object ActorSystem {
     val appConfig = bootstrapSettings.flatMap(_.config).getOrElse(ConfigFactory.load(cl))
     val executionContext = bootstrapSettings.flatMap(_.defaultExecutionContext)
 
+    val promise = Promise[Terminated]()
+    val wrappedGuardian = WhenTerminated(guardianBehavior, promise)
+    val whenTerminatedSetup = WhenTerminatedSetup(promise.future)
```

Comment:
This is a little dirty but its the only way I can see of getting the future from `wrappedGuardian` to `ActorSystemAdapter`.

Is there a more conventional of passing this into the adapter?